### ISSUE TITLE
Remove shadow buttons are unlabeled and don't show a tooltip

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#63196]
### Description

The 'Remove shadow' icon buttons in the Global Styles are unlabeled. 

More accurately, propr to the button [are passed via an object](https://github.com/WordPress/gutenberg/blob/6034e900bcb52ca6de20e298d2180977eeedcad5/packages/edit-site/src/components/global-styles/shadows-edit-panel.js#L368-L376), where an `ariaLabel` and a `tooltip` keys are set. 
However, the `Button` component either accepts a `label` prop or  an `aria-label` attribute.
Not sure why the `tooltip` prop passed this way doesn't work btut, regardless, it's unnecessary if the label prop is set correctly.

The rendered markup is invalid as the key names are rendered as HTML attributes:
- the button element has an `arialabel` attribute, it should be `aria-label`
- the button element has an `tooltip` attribute

Screenshot of the button markup in the dev tools inspector and the accessibility tab selected to show the button doesn't have an accessible name:

![Screenshot 2024-07-05 at 16 07 26](https://github.com/WordPress/gutenberg/assets/1682452/5e7f2423-3573-4cfe-b454-deed7766556b)

I'm not sure why those props are passed via an object in the first place, instead of being passed individually to the Button component. As far as I can tell there's no need to set a variable for that object in the first place.

I have to note this is one more occurrence of an unlabeled control in the editor. There have been dozens of cases because contributors to this project keep 'doing it wrong' and the base component are open to misuesd.

As pointed out in several other issues and conversations, unlabeled controls are not acceptable. The only way to make sure this doesn't happen again is by refactoring the base components and make the `label` prop required. See https://github.com/WordPress/gutenberg/issues/51740

Cc @WordPress/gutenberg-components 

### Step-by-step reproduction instructions

- Go to the Site Editor > Styles > Shadows > click one of the default shadows > click the plus icon button 'Add shadow'.
- A new shadow item will appear in the list below, with a minus icon button. Screenshot:

![Screenshot 2024-07-05 at 16 05 19](https://github.com/WordPress/gutenberg/assets/1682452/f3b89f12-5537-4ad2-bac2-f4049939dee3)

- Hover or focus the minus icon button.
- Observe no tooltip appears.
- Inspect the source and observe the button has an `arialabel` attribute instead of `aria-label`. 
- Observe there's also a `tooltip` attribute.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by mirka** [comment]
This kind of error is unfortunate because an attempt was clearly made to be accessible, and it's something that could've easily been caught if TypeScript were enabled in this section of the codebase. It seems unlikely that the Gutenberg app is going to transition to TypeScript anytime soon, so type checking is never going to be a useful strategy for upholding accessibility within Gutenberg app code 😓

I'm starting to seriously consider experimenting with runtime checks. It could be feasible if it's well-targeted, and limited to dev mode within the Gutenberg app.
